src/configuration/merge_configurations.ts

```diff
@@ -0,0 +1,50 @@
+import { IConfiguration } from './types'
+import mergeWith from 'lodash.mergewith'
+
+const ADDITIVE_ARRAYS = [
+  'format',
+  'import',
+  'name',
+  'paths',
+  'require',
+  'requireModule',
+]
+const TAG_EXPRESSIONS = ['tags', 'retryTagFilter']
+
+function mergeArrays(objValue: any[], srcValue: any[]) {
+  if (objValue && srcValue) {
+    return [].concat(objValue, srcValue)
+  }
+  return undefined
+}
+
+function mergeTagExpressions(objValue: string, srcValue: string) {
+  if (objValue && srcValue) {
+    return `${wrapTagExpression(objValue)} and ${wrapTagExpression(srcValue)}`
+  }
+  return undefined
+}
+
+function wrapTagExpression(raw: string) {
+  if (raw.startsWith('(') && raw.endsWith(')')) {
+    return raw
+  }
+  return `(${raw})`
+}
+
+function customizer(objValue: any, srcValue: any, key: string): any {
+  if (ADDITIVE_ARRAYS.includes(key)) {
+    return mergeArrays(objValue, srcValue)
+  }
+  if (TAG_EXPRESSIONS.includes(key)) {
+    return mergeTagExpressions(objValue, srcValue)
+  }
+  return undefined
+}
+
+export function mergeConfigurations<T = Partial<IConfiguration>>(
```

Comment:
This does for object-based configuration what the various helpers on `ArgvParser` do for argv-based configuration.